Resources/ProxyProvider/ProxyProviderFactory.py

```diff
@@ -0,0 +1,54 @@
+########################################################################
+# File :   ProxyProviderFactory.py
+# Author : A.T.
+########################################################################
+
+"""  The Proxy Provider Factory instantiates ProxyProvider objects
+     according to their configuration
+"""
+from DIRAC import S_OK, S_ERROR, gLogger
+from DIRAC.Core.Utilities import ObjectLoader
+from DIRAC.Resources.ProxyProvider.ProxyProvider import getProxyProviderConfigDict
+
+__RCSID__ = "$Id$"
+
+
+class ProxyProviderFactory(object):
+
+  #############################################################################
+  def __init__(self):
+    """ Standard constructor
+    """
+    self.log = gLogger.getSubLogger('ProxyProviderFactory')
```

Comment:
use `__name__` for the SubLogger
(Define a `LOG = gLogger.getSubLogger(__name__)` at the top, and use that for all logging)